utils/src/com/cloud/utils/S3Utils.java

```diff
@@ -352,10 +352,15 @@ public static File getFile(final ClientOptions clientOptions, final String bucke
         ListObjectsRequest listObjectsRequest = new ListObjectsRequest().withBucketName(bucketName).withPrefix(directory + SEPARATOR);
 
         ObjectListing ol = client.listObjects(listObjectsRequest);
-        while (ol != null && ol.isTruncated()) {
+        if(ol.isTruncated()) {
```

Comment:
is ol guaranteed not to be null? I don't see any specific remark on this in the amazon javadoc.